code/WorkInProgress/recycling/conveyor.dm

```diff
@@ -495,6 +660,40 @@
 	..()
 	update()
 
+/obj/machinery/conveyor/built/
+	desc = "A conveyor belt. This one looks like it was built recently."
+	deconstructable = TRUE
+
+	initialize() // Look around for other conveyor belts and assume a dir_in and dir_out based on that.
+		var/list/favorable_dir_ins = list()
+		var/list/favorable_dir_outs = list()
+		for (var/direction in cardinal)
+			var/turf/T = get_step(src, direction)
+			var/obj/machinery/conveyor/neighbor_conveyor = locate() in T
+
+			if (neighbor_conveyor) // If we actually have a conveyor belt in that direction...
+				if (direction == turn(neighbor_conveyor.dir_in, 180)) // Is it contrary to a dir_in of another conveyor belt?
+					favorable_dir_outs += direction // Great! A favorable dir_out.
+				if (direction == turn(neighbor_conveyor.dir_out, 180)) // Is it contrary to the dir_out of another conveyour belt?
+					favorable_dir_ins += direction // Great! A favorable dir_in.
+
+		if (length(favorable_dir_ins))
+			src.dir_in = favorable_dir_ins[1] // If there are any favorable dir ins, pick the first index.
+		if (length(favorable_dir_outs))
+			src.dir_out = favorable_dir_outs[1]
+
+		if (!src.dir_in)
+			src.dir_in = NORTH // In the case there's no favorable dir_ins, pick north.
+		if (!src.dir_out)
+			src.dir_out = turn(src.dir_in, 180) // In the case there's no favorable dir_outs, pick the contrary of dir_in.
+
+		. = ..()
```

Comment:
example: https://streamable.com/on1645

Reply:
gonna debug at home.

Reply:
The conveyors themselves seem very cool and chill to build, even if the time to screw them in is a tad long for non-engineers, but I guess that's the cost of not being an engineer. I am not sure if I agree with that trait design, but that's not really to do with this PR.^^

What seemed kind of annoying was connecting the switches to the belts, having to clickdrag for every single one seems pretty annoying? Clickdragging things with small sprites (like switches) is a relatively intensive thing to do, especially repeatedly. Also it's kind of confusing because it uses the exact same message as linking via something like mechanics connecting? This still happens if you use the Connect Component function of the multitool, even if the panel is closed.

Also, my conveyor switches seem to not be the /built subtype you created, so I am not sure if the above paragraph is a bug? Ideally I think it would be nice if the conveyor switches could be connected to the belts via the Connect Component function of the multitool as well. I am not entirely sure if this was already the intent and it's just bugged?
![image](https://github.com/goonstation/goonstation/assets/35579460/330cd0b0-3ed4-47a4-b1da-6de2977202ee)

Reply:
Also, at some point it could also be cool to allow for scanning and making the oshan carousel conveyors and carousel unit, but that can certainly be handled in a later PR. :D

Reply:
Sorry for the enormous delay, @zjdtmkhzt, but I did the few fixes. The conveyor switch thing is really a bug, Connect Component was supposed to work with conveyors.

Reply:
It's no problem to take a few days to get to making review changes!^^
I'll take a look at it later :3